### PR TITLE
Migrating spring data cosmos from sdk/cosmos to sdk/spring

### PR DESCRIPTION
**Pull request by trande4884**
# Description

We are migrating azure-spring-data-cosmos and azure-spring-data-cosmos-test from sdk/cosmos to sdk/spring.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/azure-spring-data-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * Bug fixed in `ReactiveCosmosTemplate` where returning a Flux<JsonNode> was causing an error - See [PR 33730](https://github.com/Azure/azure-sdk-for-java/pull/33730)
 
 #### Other Changes
+* The module `azure-spring-data-cosmos` was moved from sdk/cosmos to sdk/spring - See [PR 33905](https://github.com/Azure/azure-sdk-for-java/pull/33905)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-data-cosmos/CHANGELOG.md:13]: Please add similar content to the **Spring Cloud Azure**'s changelog: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/CHANGELOG.md

File: sdk/spring/pom.xml [hunk 2]
```diff
@@ -66,6 +66,8 @@
         <module>spring-cloud-azure-feature-management</module>
         <module>spring-cloud-azure-feature-management-web</module>
         <module>spring-cloud-azure-starter-appconfiguration-config</module>
+        <module>azure-spring-data-cosmos</module>
+        <module>azure-spring-data-cosmos-test</module>
```
Comment (chenrujun) [inline sdk/spring/pom.xml:70]: Please add the 2 modules into [spring-cloud-azure-dependencies](https://github.com/Azure/azure-sdk-for-java/blob/spring-cloud-azure-autoconfigure_4.6.0/sdk/boms/spring-cloud-azure-dependencies/pom.xml)

**Review by chenrujun — COMMENTED** [review]
Please update ci.yml and test.yml related contents.

> image

**Comment by trande4884** [comment]
> Please update ci.yml and test.yml related contents.
> 
> > image

@kushagraThapar do you think we want to do this? Don't we still want the azure-spring-data-cosmos stuff to run with the rest of our Java code?

File: sdk/spring/azure-spring-data-cosmos/CHANGELOG.md (hunk 1 above)
Comment (trande4884) [reply sdk/spring/azure-spring-data-cosmos/CHANGELOG.md:13]: Done.

**Comment by kushagraThapar** [comment]
> > Please update ci.yml and test.yml related contents.
> > > image
> 
> @kushagraThapar do you think we want to do this? Don't we still want the azure-spring-data-cosmos stuff to run with the rest of our Java code?

@trande4884 - yes that's correct. We want to keep this as it is.. so that we can run spring data cosmos CI with azure-cosmos code changes automatically.

**Comment by chenrujun** [comment]
Hi, @trande4884 , could you please resolve the conflicts?

File: sdk/cosmos/tests.yml [hunk 3]
```diff
@@ -50,7 +50,7 @@ stages:
           Path: sdk/cosmos/integration-matrix.json
```
Comment (benbp) [inline sdk/cosmos/tests.yml:50]:
```suggestion
          Path: sdk/spring/cosmos-integration-matrix.json
```

**Comment by chenrujun** [comment]
Hi, @trande4884 .
Could you please resolve the conflicts and fix check failures?

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml [hunk 4]
```diff
@@ -67,6 +67,16 @@
...
 
       <!-- Spring Cloud Azure -->
+      <dependency>
+        <groupId>com.azure.spring</groupId>
+        <artifactId>azure-spring-data-cosmos</artifactId>
+        <version>3.33.0-beta.1</version>
+      </dependency>
+      <dependency>
+        <groupId>com.azure.spring</groupId>
+        <artifactId>azure-spring-data-cosmos-test</artifactId>
```
Comment (chenrujun) [inline sdk/boms/spring-cloud-azure-dependencies/pom.xml:77]: This artifact is not necessary if it's only used for test, not used for release.

File: sdk/cosmos/tests.yml (hunk 3 above)
Comment (trande4884) [reply sdk/cosmos/tests.yml]: Done.

**Comment by benbp** [comment]
Pipeline/test/matrix configs all looking good to me.

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml [hunk 5]
```diff
@@ -67,6 +67,11 @@
       </dependency>
 
       <!-- Spring Cloud Azure -->
+      <dependency>
+        <groupId>com.azure.spring</groupId>
+        <artifactId>azure-spring-data-cosmos</artifactId>
+        <version>3.33.0-beta.1</version>
+      </dependency>
```
Comment (saragluna) [inline sdk/boms/spring-cloud-azure-dependencies/pom.xml:74]: Line 58 already has one for `azure-spring-data-cosmos`. Should we delete one?

Comment (trande4884) [reply sdk/boms/spring-cloud-azure-dependencies/pom.xml]: Done.

**Review by saragluna — APPROVED** [review]

**Review by saragluna — APPROVED** [review]

**Pull request merged**